diff --git a/minimatch.js b/minimatch.js
index 71c96a1fb71cce0ebca1d9d69f090032d99f44d7..bce338f2238dd6b1cb84eaa019d147877774b86f 100644
--- a/minimatch.js
+++ b/minimatch.js
@@ -119,7 +119,7 @@ const braceExpand = (pattern, options = {}) => {
   return expand(pattern)
 }
 
-const MAX_PATTERN_LENGTH = 1024 * 64
+const MAX_PATTERN_LENGTH = 1024 * 64 * 1024
 const assertValidPattern = pattern => {
   if (typeof pattern !== 'string') {
     throw new TypeError('invalid pattern')
