diff --git a/src/util/longbits.js b/src/util/longbits.js
index 11bfb1c002e29d1e91c19ac5a3696d89fbaf419c..613435a1e684a1d2ae1d541c5fd4d6e8106c6db3 100644
--- a/src/util/longbits.js
+++ b/src/util/longbits.js
@@ -112,8 +112,8 @@ LongBits.prototype.toNumber = function toNumber(unsigned) {
  * @returns {Long} Long
  */
 LongBits.prototype.toLong = function toLong(unsigned) {
-    return util.Long
-        ? new util.Long(this.lo | 0, this.hi | 0, Boolean(unsigned))
+    return (util.Long && util.Long.default)
+        ? new util.Long.default(this.lo | 0, this.hi | 0, Boolean(unsigned))
         /* istanbul ignore next */
         : { low: this.lo | 0, high: this.hi | 0, unsigned: Boolean(unsigned) };
 };
