diff --git a/node_modules/@onekeyfe/react-native-sni-connect/SniConnect.podspec b/node_modules/@onekeyfe/react-native-sni-connect/SniConnect.podspec
index 3acfb56..cd05d30 100644
--- a/node_modules/@onekeyfe/react-native-sni-connect/SniConnect.podspec
+++ b/node_modules/@onekeyfe/react-native-sni-connect/SniConnect.podspec
@@ -24,7 +24,7 @@ Pod::Spec.new do |s|
   s.dependency 'React-jsi'
   s.dependency 'React-callinvoker'
   s.dependency 'React-Codegen'
-  s.dependency 'EMASCurl', '1.4.1'
+  s.dependency 'EMASCurl', '1.5.5'
 
   s.pod_target_xcconfig = {
     "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/Headers/Private/React-Core\"",
