
diff --git a/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm b/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.m
similarity index 97%
rename from node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm
rename to node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.m
index 508494a..448401b 100644
--- a/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm
+++ b/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.m
@@ -641,6 +641,28 @@ - (NSDictionary *)_writeEntry:(NSArray<NSString *> *)entry changedManifest:(BOOL
     return errorOut;
 }
 
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getValueForKey:(NSString *)key)
+{
+    
+    NSString *value =
+        _manifest[key];  // nil means missing, null means there may be a data file, else: NSString
+    if (value == (id)kCFNull) {
+        value = [RCTGetCache() objectForKey:key];
+        if (!value) {
+            NSString *filePath = [self _filePathForKey:key];
+            value = RCTReadFile(filePath, key, nil);
+            if (value) {
+                [RCTGetCache() setObject:value forKey:key cost:value.length];
+            } else {
+                // file does not exist after all, so remove from manifest (no need to save
+                // manifest immediately though, as cost of checking again next time is negligible)
+                [_manifest removeObjectForKey:key];
+            }
+        }
+    }
+    return value;
+}
+
 - (void)_multiGet:(NSArray<NSString *> *)keys
          callback:(RCTResponseSenderBlock)callback
            getter:(NSString * (^)(NSUInteger i, NSString *key, NSDictionary **errorOut))getValue

 