diff --git a/node_modules/@reown/appkit-scaffold-react-native/lib/commonjs/modal/w3m-modal/styles.js b/node_modules/@reown/appkit-scaffold-react-native/lib/commonjs/modal/w3m-modal/styles.js
index 4408fba..a43b38c 100644
--- a/node_modules/@reown/appkit-scaffold-react-native/lib/commonjs/modal/w3m-modal/styles.js
+++ b/node_modules/@reown/appkit-scaffold-react-native/lib/commonjs/modal/w3m-modal/styles.js
@@ -8,7 +8,9 @@ var _reactNative = require("react-native");
 var _default = exports.default = _reactNative.StyleSheet.create({
   modal: {
     margin: 0,
-    justifyContent: 'flex-end'
+    justifyContent: 'flex-end',
+    zIndex: 999999,
+    elevation: 999999
   },
   card: {
     borderBottomLeftRadius: 0,
diff --git a/node_modules/@reown/appkit-scaffold-react-native/lib/module/modal/w3m-modal/styles.js b/node_modules/@reown/appkit-scaffold-react-native/lib/module/modal/w3m-modal/styles.js
index 08ead2c..68e8e85 100644
--- a/node_modules/@reown/appkit-scaffold-react-native/lib/module/modal/w3m-modal/styles.js
+++ b/node_modules/@reown/appkit-scaffold-react-native/lib/module/modal/w3m-modal/styles.js
@@ -2,7 +2,9 @@ import { StyleSheet } from 'react-native';
 export default StyleSheet.create({
   modal: {
     margin: 0,
-    justifyContent: 'flex-end'
+    justifyContent: 'flex-end',
+    zIndex: 999999,
+    elevation: 999999
   },
   card: {
     borderBottomLeftRadius: 0,
