diff --git a/node_modules/@sentry/browser/build/npm/cjs/dev/integrations/breadcrumbs.js b/node_modules/@sentry/browser/build/npm/cjs/dev/integrations/breadcrumbs.js
index 939142d..6be3c81 100644
--- a/node_modules/@sentry/browser/build/npm/cjs/dev/integrations/breadcrumbs.js
+++ b/node_modules/@sentry/browser/build/npm/cjs/dev/integrations/breadcrumbs.js
@@ -189,7 +189,7 @@ function _getXhrBreadcrumbHandler(client) {
       return;
     }
 
-    const { method, url, status_code, body } = sentryXhrData;
+    const { method, url, status_code, body, request_headers = {} } = sentryXhrData;
 
     const data = {
       method,
@@ -197,6 +197,10 @@ function _getXhrBreadcrumbHandler(client) {
       status_code,
     };
 
+    if (request_headers['x-onekey-request-id']) {
+      data.request_id = request_headers['x-onekey-request-id'];
+    }
+
     const hint = {
       xhr: handlerData.xhr,
       input: body,
diff --git a/node_modules/@sentry/browser/build/npm/cjs/prod/integrations/breadcrumbs.js b/node_modules/@sentry/browser/build/npm/cjs/prod/integrations/breadcrumbs.js
index 939142d..6be3c81 100644
--- a/node_modules/@sentry/browser/build/npm/cjs/prod/integrations/breadcrumbs.js
+++ b/node_modules/@sentry/browser/build/npm/cjs/prod/integrations/breadcrumbs.js
@@ -189,7 +189,7 @@ function _getXhrBreadcrumbHandler(client) {
       return;
     }
 
-    const { method, url, status_code, body } = sentryXhrData;
+    const { method, url, status_code, body, request_headers = {} } = sentryXhrData;
 
     const data = {
       method,
@@ -197,6 +197,10 @@ function _getXhrBreadcrumbHandler(client) {
       status_code,
     };
 
+    if (request_headers['x-onekey-request-id']) {
+      data.request_id = request_headers['x-onekey-request-id'];
+    }
+
     const hint = {
       xhr: handlerData.xhr,
       input: body,
diff --git a/node_modules/@sentry/browser/build/npm/esm/dev/integrations/breadcrumbs.js b/node_modules/@sentry/browser/build/npm/esm/dev/integrations/breadcrumbs.js
index bd15e0b..c317bc7 100644
--- a/node_modules/@sentry/browser/build/npm/esm/dev/integrations/breadcrumbs.js
+++ b/node_modules/@sentry/browser/build/npm/esm/dev/integrations/breadcrumbs.js
@@ -187,7 +187,7 @@ function _getXhrBreadcrumbHandler(client) {
       return;
     }
 
-    const { method, url, status_code, body } = sentryXhrData;
+    const { method, url, status_code, body, request_headers = {} } = sentryXhrData;
 
     const data = {
       method,
@@ -195,6 +195,10 @@ function _getXhrBreadcrumbHandler(client) {
       status_code,
     };
 
+    if (request_headers['x-onekey-request-id']) {
+      data.request_id = request_headers['x-onekey-request-id'];
+    }
+
     const hint = {
       xhr: handlerData.xhr,
       input: body,
diff --git a/node_modules/@sentry/browser/build/npm/esm/prod/integrations/breadcrumbs.js b/node_modules/@sentry/browser/build/npm/esm/prod/integrations/breadcrumbs.js
index bd15e0b..c317bc7 100644
--- a/node_modules/@sentry/browser/build/npm/esm/prod/integrations/breadcrumbs.js
+++ b/node_modules/@sentry/browser/build/npm/esm/prod/integrations/breadcrumbs.js
@@ -187,7 +187,7 @@ function _getXhrBreadcrumbHandler(client) {
       return;
     }
 
-    const { method, url, status_code, body } = sentryXhrData;
+    const { method, url, status_code, body, request_headers = {} } = sentryXhrData;
 
     const data = {
       method,
@@ -195,6 +195,10 @@ function _getXhrBreadcrumbHandler(client) {
       status_code,
     };
 
+    if (request_headers['x-onekey-request-id']) {
+      data.request_id = request_headers['x-onekey-request-id'];
+    }
+
     const hint = {
       xhr: handlerData.xhr,
       input: body,
