diff --git a/node_modules/@starcoin/starcoin/dist/starcoin.js b/node_modules/@starcoin/starcoin/dist/starcoin.js
index cd80f13..80fb4b9 100644
--- a/node_modules/@starcoin/starcoin/dist/starcoin.js
+++ b/node_modules/@starcoin/starcoin/dist/starcoin.js
@@ -438,8 +438,8 @@ var BinarySerializer = /*#__PURE__*/function () {
     var low = BigInt(value) & BinarySerializer.BIG_32Fs;
     var high = BigInt(value) >> BinarySerializer.BIG_32; // write little endian number
 
-    this.serializeU32(Number(low));
-    this.serializeU32(Number(high));
+    this.serializeU32(parseInt(low));
+    this.serializeU32(parseInt(high));
   };
 
   _proto.serializeU128 = function serializeU128(value) {
@@ -475,8 +475,8 @@ var BinarySerializer = /*#__PURE__*/function () {
     var low = BigInt(value) & BinarySerializer.BIG_32Fs;
     var high = BigInt(value) >> BinarySerializer.BIG_32; // write little endian number
 
-    this.serializeI32(Number(low));
-    this.serializeI32(Number(high));
+    this.serializeI32(parseInt(low));
+    this.serializeI32(parseInt(high));
   };
 
   _proto.serializeI128 = function serializeI128(value) {
