diff --git a/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js b/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js
index 47addac..3b32c89 100644
--- a/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js
+++ b/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js
@@ -160,11 +160,60 @@ function unlisten() {
   }), dispose.clear();
 }
 var setupVersion = -1;
+
+
+const ExpoDevice = require('expo-device');
+function isTabletOrAndroid(callback) {
+  if (import_constants.isAndroid) {
+    callback(true);
+  } else if (ExpoDevice.deviceType === ExpoDevice.DeviceType.TABLET) {
+    callback(true);
+  } else {
+    callback(false);
+  }
+}
+
 function setupMediaListeners() {
   var _loop = function(key2) {
     var update = function() {
       var next = !!getMatch().matches;
-      next !== mediaState[key2] && (mediaState = _object_spread_props(_object_spread({}, mediaState), _define_property({}, key2, next)), updateCurrentState());
+      isTabletOrAndroid((_result5) => {
+        if (_result5) {
+           switch (key2) {
+            case 'md':
+              next = true;
+              break;
+            case 'gtMd':
+              next = false;
+              break;
+            case '2md':
+              next = false;
+              break;
+            case 'gt2md':
+              next = false;
+              break;
+            case 'lg':
+              next = false;
+              break;
+            case 'gtLg':
+              next = false;
+              break;
+            case 'xl':
+              next = false;
+              break;
+            case 'gtXl':
+              next = false;
+              break;
+            case '2xl':
+              next = false;
+              break;
+            case 'gt2xl':
+              next = false;
+              break;
+           }
+        }
+        next !== mediaState[key2] && (mediaState = _object_spread_props(_object_spread({}, mediaState), _define_property({}, key2, next)), updateCurrentState());
+      })
     }, str = mediaObjectToString(mediaQueryConfig[key2], key2), getMatch = function() {
       return (0, import_matchMedia.matchMedia)(str);
     }, match = getMatch();
