diff --git a/node_modules/@tamagui/portal/src/Portal.native.tsx b/node_modules/@tamagui/portal/src/Portal.native.tsx
--- a/node_modules/@tamagui/portal/src/Portal.native.tsx
+++ b/node_modules/@tamagui/portal/src/Portal.native.tsx
@@ -6,13 +6,8 @@ import { Platform, RootTagContext } from 'react-native'
 import { PortalItem } from './GorhomPortal'
 import type { PortalProps } from './PortalProps'
 
-const isFabric = global?.nativeFabricUIManager
-let createPortal
-if (isFabric) {
-  createPortal = require('react-native/Libraries/Renderer/shims/ReactFabric').createPortal
-} else {
-  createPortal = require('react-native/Libraries/Renderer/shims/ReactNative').createPortal
-}
+const createPortal =
+  require('react-native/Libraries/Renderer/shims/ReactFabric').createPortal
 
 export const Portal = (props: PortalProps) => {
   const rootTag = React.useContext(RootTagContext)
diff --git a/node_modules/@tamagui/portal/dist/cjs/Portal.native.js b/node_modules/@tamagui/portal/dist/cjs/Portal.native.js
--- a/node_modules/@tamagui/portal/dist/cjs/Portal.native.js
+++ b/node_modules/@tamagui/portal/dist/cjs/Portal.native.js
@@ -46,8 +46,7 @@ function _object_spread(target) {
   }
   return target;
 }
-var _global, isFabric = (_global = global) === null || _global === void 0 ? void 0 : _global.nativeFabricUIManager, createPortal;
-isFabric ? createPortal = require("react-native/Libraries/Renderer/shims/ReactFabric").createPortal : createPortal = require("react-native/Libraries/Renderer/shims/ReactNative").createPortal;
+var createPortal = require("react-native/Libraries/Renderer/shims/ReactFabric").createPortal;
 var Portal = function(props) {
   var rootTag = React.useContext(import_react_native.RootTagContext), contents = /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_stacks.YStack, _object_spread({
     pointerEvents: "box-none",
diff --git a/node_modules/@tamagui/portal/dist/esm/Portal.native.js b/node_modules/@tamagui/portal/dist/esm/Portal.native.js
--- a/node_modules/@tamagui/portal/dist/esm/Portal.native.js
+++ b/node_modules/@tamagui/portal/dist/esm/Portal.native.js
@@ -22,8 +22,7 @@ function _object_spread(target) {
   }
   return target;
 }
-var _global, isFabric = (_global = global) === null || _global === void 0 ? void 0 : _global.nativeFabricUIManager, createPortal;
-isFabric ? createPortal = require("react-native/Libraries/Renderer/shims/ReactFabric").createPortal : createPortal = require("react-native/Libraries/Renderer/shims/ReactNative").createPortal;
+var createPortal = require("react-native/Libraries/Renderer/shims/ReactFabric").createPortal;
 var Portal = function(props) {
   var rootTag = React.useContext(RootTagContext), contents = /* @__PURE__ */ _jsx(YStack, _object_spread({
     pointerEvents: "box-none",
