diff --git a/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.js b/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.js
index 956ff30..9e45309 100644
--- a/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.js
+++ b/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.js
@@ -64,7 +64,8 @@ const configureMedia = (config) => {
     mediaVersion++;
     for (const key in media)
       mediaState[key] = (mediaQueryDefaultActive == null ? void 0 : mediaQueryDefaultActive[key]) || !1, mediaKeys.add(`$${key}`);
-    Object.assign(mediaQueryConfig, media), initState = { ...mediaState }, mediaKeysOrdered = Object.keys(media), config.disableSSR ? setupMediaListeners() : updateCurrentState();
+    // We moved `initState = { ...mediaState }`` to the end because `setupMediaListeners.update`` will refresh `mediaState`, and we can get the correct initial values for `initState`.
+    Object.assign(mediaQueryConfig, media), mediaKeysOrdered = Object.keys(media), config.disableSSR ? setupMediaListeners() : updateCurrentState(), initState = { ...mediaState };
   }
 };
 function unlisten() {
diff --git a/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js b/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js
index 080caf2..fbfba93 100644
--- a/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js
+++ b/node_modules/@tamagui/web/dist/cjs/hooks/useMedia.native.js
@@ -65,7 +65,8 @@ const configureMedia = (config) => {
     mediaVersion++;
     for (const key in media)
       mediaState[key] = (mediaQueryDefaultActive == null ? void 0 : mediaQueryDefaultActive[key]) || !1, mediaKeys.add(`$${key}`);
-    Object.assign(mediaQueryConfig, media), initState = { ...mediaState }, mediaKeysOrdered = Object.keys(media), config.disableSSR ? setupMediaListeners() : updateCurrentState();
+    // We moved `initState = { ...mediaState }`` to the end because `setupMediaListeners.update`` will refresh `mediaState`, and we can get the correct initial values for `initState`.
+    Object.assign(mediaQueryConfig, media), mediaKeysOrdered = Object.keys(media), config.disableSSR ? setupMediaListeners() : updateCurrentState(), initState = { ...mediaState };
   }
 };
 function unlisten() {
diff --git a/node_modules/@tamagui/web/dist/esm/hooks/useMedia.js b/node_modules/@tamagui/web/dist/esm/hooks/useMedia.js
index 4bb9787..a5ca05d 100644
--- a/node_modules/@tamagui/web/dist/esm/hooks/useMedia.js
+++ b/node_modules/@tamagui/web/dist/esm/hooks/useMedia.js
@@ -34,7 +34,8 @@ const configureMedia = (config) => {
     mediaVersion++;
     for (const key in media)
       mediaState[key] = mediaQueryDefaultActive?.[key] || !1, mediaKeys.add(`$${key}`);
-    Object.assign(mediaQueryConfig, media), initState = { ...mediaState }, mediaKeysOrdered = Object.keys(media), config.disableSSR ? setupMediaListeners() : updateCurrentState();
+    // We moved `initState = { ...mediaState }`` to the end because `setupMediaListeners.update`` will refresh `mediaState`, and we can get the correct initial values for `initState`.
+    Object.assign(mediaQueryConfig, media), mediaKeysOrdered = Object.keys(media), config.disableSSR ? setupMediaListeners() : updateCurrentState(), initState = { ...mediaState };
   }
 };
 function unlisten() {
