diff --git a/node_modules/@tamagui/animations-moti/dist/cjs/createAnimations.native.js b/node_modules/@tamagui/animations-moti/dist/cjs/createAnimations.native.js
index 43e8d14..7b4694d 100644
--- a/node_modules/@tamagui/animations-moti/dist/cjs/createAnimations.native.js
+++ b/node_modules/@tamagui/animations-moti/dist/cjs/createAnimations.native.js
@@ -65,7 +65,9 @@ function createAnimations(animations) {
       return (0, import_react_native_reanimated.useAnimatedReaction)(
         () => instance.value,
         (next, prev) => {
-          prev !== next && (0, import_react_native_reanimated.runOnJS)(onValue)(next);
+          // I suspect that the crash is caused by an issue with Babel transpilation. I've temporarily commented it out to prevent the UI from crashing
+          // 
+          // prev !== next && (0, import_react_native_reanimated.runOnJS)(onValue)(next);
         },
         // dependency array is very important here
         [onValue, instance]
