diff --git a/node_modules/babel-preset-expo/index.js b/node_modules/babel-preset-expo/index.js
index 2099ee3..423182c 100644
--- a/node_modules/babel-preset-expo/index.js
+++ b/node_modules/babel-preset-expo/index.js
@@ -39,6 +39,24 @@ module.exports = function (api, options = {}) {
     );
   }
 
+   // TODO: remove this on Expo 50.x
+  // fix(babel): fix unused platform code removal #25171
+  // https://github.com/expo/expo/pull/25171
+  if (require('@babel/plugin-transform-parameters')) {
+    // Metro applies this plugin too but it does it after the imports have been transformed which breaks
+    // the plugin. Here, we'll apply it before the commonjs transform, in production, to ensure `Platform.OS`
+    // is replaced with a string literal and `__DEV__` is converted to a boolean.
+    // Applying early also means that web can be transformed before the `react-native-web` transform mutates the import.
+    extraPlugins.push([
+        require('metro-transform-plugins/src/inline-plugin.js'),
+        {
+            dev: false,
+            inlinePlatform: true,
+            platform,
+        },
+    ]);
+  }
+
   // Set true to disable `@babel/plugin-transform-react-jsx`
   // we override this logic outside of the metro preset so we can add support for
   // React 17 automatic JSX transformations.
@@ -105,7 +123,8 @@ module.exports = function (api, options = {}) {
       ],
     ],
     plugins: [
-      getObjectRestSpreadPlugin(),
+      // disable object rest spread plugin because hermes is enabled by default.
+      // getObjectRestSpreadPlugin(),
       ...extraPlugins,
       getAliasPlugin(),
       [require.resolve('@babel/plugin-proposal-decorators'), { legacy: true }],
