diff --git a/node_modules/big-integer/BigInteger.js b/node_modules/big-integer/BigInteger.js
index 605a498..eea0594 100644
--- a/node_modules/big-integer/BigInteger.js
+++ b/node_modules/big-integer/BigInteger.js
@@ -1230,6 +1230,9 @@ var bigInt = (function (undefined) {
     SmallInteger.prototype.toJSNumber = SmallInteger.prototype.valueOf;
 
     function parseStringValue(v) {
+        if (v.startsWith("0x") || v.startsWith("0X")) {
+            return bigInt(v.slice(2), 16);
+        }
         if (isPrecise(+v)) {
             var x = +v;
             if (x === truncate(x))
