diff --git a/node_modules/console-feed/lib/Transform/index.d.ts b/node_modules/console-feed/lib/Transform/index.d.ts
index 2e2f031..dff62ee 100644
--- a/node_modules/console-feed/lib/Transform/index.d.ts
+++ b/node_modules/console-feed/lib/Transform/index.d.ts
@@ -1,3 +1,5 @@
 import { Message } from '../definitions/Console';
+import Replicator from './replicator'
 export declare function Encode<T>(data: any): T;
 export declare function Decode(data: any): Message;
+export declare const replicator: Replicator;
diff --git a/node_modules/console-feed/lib/Transform/index.js b/node_modules/console-feed/lib/Transform/index.js
index fe5b826..c6a8d85 100644
--- a/node_modules/console-feed/lib/Transform/index.js
+++ b/node_modules/console-feed/lib/Transform/index.js
@@ -3,7 +3,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
     return (mod && mod.__esModule) ? mod : { "default": mod };
 };
 exports.__esModule = true;
-exports.Decode = exports.Encode = void 0;
+exports.Decode = exports.Encode = exports.replicator = void 0;
 var arithmetic_1 = __importDefault(require("./arithmetic"));
 var Function_1 = __importDefault(require("./Function"));
 var HTML_1 = __importDefault(require("./HTML"));
@@ -20,4 +20,5 @@ function Decode(data) {
     return replicator.decode(JSON.stringify(data));
 }
 exports.Decode = Decode;
+exports.replicator=replicator;
 //# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi9zcmMvVHJhbnNmb3JtL2luZGV4LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7Ozs7OztBQUNBLDREQUFxQztBQUNyQyx3REFBaUM7QUFDakMsZ0RBQXlCO0FBQ3pCLDhDQUF1QjtBQUV2Qiw0REFBcUM7QUFFckMsSUFBTSxVQUFVLEdBQUcsQ0FBQyxpQkFBSSxFQUFFLHFCQUFRLEVBQUUsdUJBQVUsRUFBRSxnQkFBRyxDQUFDLENBQUE7QUFFcEQsSUFBTSxVQUFVLEdBQUcsSUFBSSx1QkFBVSxFQUFFLENBQUE7QUFDbkMsVUFBVSxDQUFDLGFBQWEsQ0FBQyxVQUFVLENBQUMsQ0FBQTtBQUVwQyxTQUFnQixNQUFNLENBQUksSUFBUztJQUNqQyxPQUFPLElBQUksQ0FBQyxLQUFLLENBQUMsVUFBVSxDQUFDLE1BQU0sQ0FBQyxJQUFJLENBQUMsQ0FBQyxDQUFBO0FBQzVDLENBQUM7QUFGRCx3QkFFQztBQUVELFNBQWdCLE1BQU0sQ0FBQyxJQUFTO0lBQzlCLE9BQU8sVUFBVSxDQUFDLE1BQU0sQ0FBQyxJQUFJLENBQUMsU0FBUyxDQUFDLElBQUksQ0FBQyxDQUFDLENBQUE7QUFDaEQsQ0FBQztBQUZELHdCQUVDIn0=
