diff --git a/node_modules/electron-updater/out/AppUpdater.d.ts b/node_modules/electron-updater/out/AppUpdater.d.ts
index 62cb06d..854c4e5 100644
--- a/node_modules/electron-updater/out/AppUpdater.d.ts
+++ b/node_modules/electron-updater/out/AppUpdater.d.ts
@@ -15,6 +15,7 @@ export type AppUpdaterEvents = {
     error: (error: Error, message?: string) => void;
     login: (info: AuthInfo, callback: LoginCallback) => void;
     "checking-for-update": () => void;
+    "update-download-fileInfo": (fileInfo: ResolvedUpdateFileInfo) => void;
     "update-not-available": (info: UpdateInfo) => void;
     "update-available": (info: UpdateInfo) => void;
     "update-downloaded": (event: UpdateDownloadedEvent) => void;
diff --git a/node_modules/electron-updater/out/AppUpdater.js b/node_modules/electron-updater/out/AppUpdater.js
index 0200f4a..552879e 100644
--- a/node_modules/electron-updater/out/AppUpdater.js
+++ b/node_modules/electron-updater/out/AppUpdater.js
@@ -489,6 +489,7 @@ class AppUpdater extends events_1.EventEmitter {
     }
     async executeDownload(taskOptions) {
         const fileInfo = taskOptions.fileInfo;
+        this.emit("update-download-fileInfo", taskOptions.fileInfo);
         const downloadOptions = {
             headers: taskOptions.downloadUpdateOptions.requestHeaders,
             cancellationToken: taskOptions.downloadUpdateOptions.cancellationToken,
