diff --git a/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt b/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt
index 819631b..a4126fd 100644
--- a/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt
+++ b/node_modules/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt
@@ -5,6 +5,7 @@ import expo.modules.devlauncher.helpers.await
 import expo.modules.devlauncher.helpers.fetch
 import expo.modules.manifests.core.Manifest
 import okhttp3.Headers
+import okhttp3.Headers.Companion.toHeaders
 import okhttp3.OkHttpClient
 import org.json.JSONObject
 import java.io.Reader
@@ -38,6 +39,6 @@ class DevLauncherManifestParser(
   }
 
   private fun getHeaders(): Headers {
-    return Headers.of(mapOf("expo-platform" to "android"))
+    return mapOf("expo-platform" to "android").toHeaders()
   }
 }
