diff --git a/node_modules/expo/bin/cli.js b/node_modules/expo/bin/cli.js
index 9bd972e..acce94d 100755
--- a/node_modules/expo/bin/cli.js
+++ b/node_modules/expo/bin/cli.js
@@ -7,7 +7,7 @@ run();
 
 function run() {
   // Use new beta CLI.
-  if (boolish('EXPO_USE_BETA_CLI', false)) {
+  if (boolish('EXPO_USE_BETA_CLI', true)) {
     return spawn(require.resolve('@expo/cli'), process.argv.slice(2), { stdio: 'inherit' }).on(
       'exit',
       (code) => {
