diff --git a/node_modules/expo-image/build/index.d.ts b/node_modules/expo-image/build/index.d.ts
index ce18dad..4043e77 100644
--- a/node_modules/expo-image/build/index.d.ts
+++ b/node_modules/expo-image/build/index.d.ts
@@ -2,4 +2,5 @@ export * from './Image.types';
 export { Image } from './Image';
 export { ImageBackground } from './ImageBackground';
 export { useImage } from './useImage';
+export { resolveSource, resolveSources } from './utils/resolveSources';
 //# sourceMappingURL=index.d.ts.map
\ No newline at end of file
diff --git a/node_modules/expo-image/ios/ExpoImage.podspec b/node_modules/expo-image/ios/ExpoImage.podspec
index 818e050..271ae26 100644
--- a/node_modules/expo-image/ios/ExpoImage.podspec
+++ b/node_modules/expo-image/ios/ExpoImage.podspec
@@ -20,10 +20,10 @@ Pod::Spec.new do |s|
 
   s.dependency 'ExpoModulesCore'
   s.dependency 'SDWebImage', '~> 5.21.0'
-  s.dependency 'SDWebImageAVIFCoder', '~> 0.11.0'
+  # s.dependency 'SDWebImageAVIFCoder', '~> 0.11.0'
   s.dependency 'SDWebImageSVGCoder', '~> 1.7.0'
   s.dependency 'SDWebImageWebPCoder', '~> 0.14.6'
-  s.dependency 'libavif/libdav1d'
+  # s.dependency 'libavif/libdav1d'
 
   # Swift/Objective-C compatibility
   s.pod_target_xcconfig = {
diff --git a/node_modules/expo-image/ios/ImageModule.swift b/node_modules/expo-image/ios/ImageModule.swift
index b44f431..9a51548 100644
--- a/node_modules/expo-image/ios/ImageModule.swift
+++ b/node_modules/expo-image/ios/ImageModule.swift
@@ -2,7 +2,7 @@
 
 import ExpoModulesCore
 import SDWebImage
-import SDWebImageAVIFCoder
+// import SDWebImageAVIFCoder
 import SDWebImageSVGCoder
 
 public final class ImageModule: Module {
@@ -251,8 +251,9 @@ public final class ImageModule: Module {
 
   static func registerCoders() {
     SDImageCodersManager.shared.addCoder(WebPCoder.shared)
-    SDImageCodersManager.shared.addCoder(SDImageAVIFCoder.shared)
-    SDImageCodersManager.shared.addCoder(SDImageSVGCoder.shared)
+    // SDImageCodersManager.shared.addCoder(SDImageAVIFCoder.shared)
+    // uncaught exception 'NSInternalInconsistencyException', reason: 'UIGraphicsBeginImageContext() failed to allocate CGBitampContext
+    // SDImageCodersManager.shared.addCoder(SDImageSVGCoder.shared)
     SDImageCodersManager.shared.addCoder(SDImageHEICCoder.shared)
   }
 
diff --git a/node_modules/expo-image/src/index.ts b/node_modules/expo-image/src/index.ts
index 6fa0066..858ae95 100644
--- a/node_modules/expo-image/src/index.ts
+++ b/node_modules/expo-image/src/index.ts
@@ -2,3 +2,4 @@ export * from './Image.types';
 export { Image } from './Image';
 export { ImageBackground } from './ImageBackground';
 export { useImage } from './useImage';
+export { resolveSource, resolveSources } from './utils/resolveSources';
\ No newline at end of file
