diff --git a/node_modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt b/node_modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
index edc70f7..6074f5a 100644
--- a/node_modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
+++ b/node_modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
@@ -6,6 +6,7 @@ import android.text.TextUtils
 import android.content.Context
 import android.os.Build.VERSION
 import android.os.Build.VERSION_CODES
+import android.content.res.Resources;
 
 import expo.modules.core.Promise
 import expo.modules.core.ExportedModule
@@ -63,8 +64,7 @@ class LocalizationModule(context: Context) : ExportedModule(context) {
 
   private val locales: ArrayList<Locale>
     get() {
-      val context = applicationContext ?: return ArrayList()
-      val configuration = context.resources.configuration
+      val configuration = Resources.getSystem().configuration
       return if (VERSION.SDK_INT > VERSION_CODES.N) {
         val locales = ArrayList<Locale>()
         for (i in 0 until configuration.locales.size()) {
