diff --git a/node_modules/keytar/binding.gyp b/node_modules/keytar/binding.gyp
index 9be3dcb..df66b96 100644
--- a/node_modules/keytar/binding.gyp
+++ b/node_modules/keytar/binding.gyp
@@ -7,7 +7,18 @@
       ],
       'cflags!': [ '-fno-exceptions' ],
       'cflags_cc!': [ '-fno-exceptions' ],
-      'xcode_settings': { 'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
+      'xcode_settings': {
+        'OTHER_CFLAGS': [
+          '-arch x86_64',
+          '-arch arm64'
+        ],
+        'OTHER_LDFLAGS': [
+          '-Wl, -bind_at_load',
+          '-framework CoreFoundation -framework CoreServices',
+          '-arch x86_64',
+          '-arch arm64'
+        ],
+        'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
         'CLANG_CXX_LIBRARY': 'libc++',
         'MACOSX_DEPLOYMENT_TARGET': '10.7',
       },
