diff --git a/node_modules/metro/.DS_Store b/node_modules/metro/.DS_Store
new file mode 100644
index 0000000..5172429
Binary files /dev/null and b/node_modules/metro/.DS_Store differ
diff --git a/node_modules/metro/src/Bundler.js b/node_modules/metro/src/Bundler.js
index eeca6ef..5c930a6 100644
--- a/node_modules/metro/src/Bundler.js
+++ b/node_modules/metro/src/Bundler.js
@@ -15,6 +15,7 @@ const Transformer = require("./DeltaBundler/Transformer");
 const DependencyGraph = require("./node-haste/DependencyGraph");
 class Bundler {
   constructor(config, options) {
+    this._config = config;
     this._depGraph = new DependencyGraph(config, options);
     this._readyPromise = this._depGraph
       .ready()
@@ -44,6 +45,9 @@ class Bundler {
     await this._depGraph.ready();
     this._transformer.end();
     this._depGraph.end();
+    if (this._config && this._config.hooks && this._config.hooks.onEnd) {
+      await this._config.hooks.onEnd();
+    }
   }
   async getDependencyGraph() {
     await this._depGraph.ready();
