diff --git a/node_modules/metro/src/Server.js b/node_modules/metro/src/Server.js
index 6ca8748..8878300 100644
--- a/node_modules/metro/src/Server.js
+++ b/node_modules/metro/src/Server.js
@@ -939,7 +939,7 @@ class Server {
           file !== null && file !== void 0 ? file : ""
         );
 
-        if (collapse || lineNumber == null || urls.has(fileAbsolute)) {
+        if (collapse || lineNumber == null || urls.has(fileAbsolute) || file?.startsWith?.('http')) {
           continue;
         }
 
