diff --git a/node_modules/react-native-screens/ios/RNSScreenStack.mm b/node_modules/react-native-screens/ios/RNSScreenStack.mm
index 498b81d..3cde0f9 100644
--- a/node_modules/react-native-screens/ios/RNSScreenStack.mm
+++ b/node_modules/react-native-screens/ios/RNSScreenStack.mm
@@ -858,13 +858,17 @@ - (void)rnscreens_disableInteractions
 {
   // When transitioning between screens, disable interactions on stack subview which wraps the screens
   // and sink all gesture events. This should work for nested stacks and stack inside bottom tabs, inside stack.
-  self.subviews[0].userInteractionEnabled = NO;
+  if (self.subviews.count > 0) {
+    self.subviews[0].userInteractionEnabled = NO;
+  }
   [self addGestureRecognizer:_sinkEventsPanGestureRecognizer];
 }
 
 - (void)rnscreens_enableInteractions
 {
-  self.subviews[0].userInteractionEnabled = YES;
+  if (self.subviews.count > 0) {
+    self.subviews[0].userInteractionEnabled = YES;
+  }
   [self removeGestureRecognizer:_sinkEventsPanGestureRecognizer];
 }
 
