diff --git a/node_modules/react-native-vision-camera/android/build.gradle b/node_modules/react-native-vision-camera/android/build.gradle
index 725696c..ab89aa0 100644
--- a/node_modules/react-native-vision-camera/android/build.gradle
+++ b/node_modules/react-native-vision-camera/android/build.gradle
@@ -229,7 +229,7 @@ dependencies {
     extractJNI("com.facebook.fbjni:fbjni:+")
 
     def rnAAR = fileTree("${nodeModules}/react-native/android").matching({ it.include "**/**/*.aar" }).singleFile
-    def jscAAR = fileTree("${nodeModules}/jsc-android/dist/org/webkit/android-jsc").matching({ it.include "**/**/*.aar" }).singleFile
+    def jscAAR = fileTree("${nodeModules}/jsc-android/dist/org/webkit/android-jsc-intl").matching({ it.include "**/**/*.aar" }).singleFile
 
     def inputFile = file("${nodeModules}/react-native/package.json")
     def json = new JsonSlurper().parseText(inputFile.text)
@@ -430,7 +430,7 @@ if (ENABLE_FRAME_PROCESSORS) {
         throw new GradleScriptException("The jsc-android npm package is missing its \"dist\" directory", null)
       }
 
-      def jscAAR = fileTree(jscDist).matching({ it.include "**/android-jsc/**/*.aar" }).singleFile
+      def jscAAR = fileTree(jscDist).matching({ it.include "**/android-jsc-intl/**/*.aar" }).singleFile
       def soFiles = zipTree(jscAAR).matching({ it.include "**/*.so" })
 
       def headerFiles = fileTree(jscDist).matching({ it.include "**/include/*.h" })
