diff --git a/node_modules/react-native-crypto/index.js b/node_modules/react-native-crypto/index.js
index f644543..bf38ebb 100644
--- a/node_modules/react-native-crypto/index.js
+++ b/node_modules/react-native-crypto/index.js
@@ -1,28 +1,16 @@
 'use strict'
+import { getRandomValues } from 'react-native-get-random-values';
 
-import { randomBytes } from 'react-native-randombytes'
-exports.randomBytes = exports.rng = exports.pseudoRandomBytes = exports.prng = randomBytes
-
-// implement window.getRandomValues(), for packages that rely on it
-if (typeof window === 'object') {
-  if (!window.crypto) window.crypto = {}
-  if (!window.crypto.getRandomValues) {
-    window.crypto.getRandomValues = function getRandomValues (arr) {
-      let orig = arr
-      if (arr.byteLength != arr.length) {
-        // Get access to the underlying raw bytes
-        arr = new Uint8Array(arr.buffer)
-      }
-      const bytes = randomBytes(arr.length)
-      for (var i = 0; i < bytes.length; i++) {
-        arr[i] = bytes[i]
-      }
-
-      return orig
-    }
+if (typeof globalThis === 'object') {
+  if (!globalThis.crypto) {
+    globalThis.crypto = {};
+  }
+  if (!globalThis.crypto.getRandomValues) {
+    globalThis.crypto.getRandomValues = getRandomValues;
   }
 }
 
+
 exports.createHash = exports.Hash = require('create-hash')
 exports.createHmac = exports.Hmac = require('create-hmac')
 
