diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt
index cf44d7d..4a8af44 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt
@@ -2,6 +2,7 @@ package com.swmansion.rnscreens
 
 import android.content.Context
 import android.content.ContextWrapper
+import android.view.MotionEvent
 import android.view.View
 import android.view.ViewGroup
 import android.view.ViewParent
@@ -16,6 +17,7 @@ import com.facebook.react.modules.core.ChoreographerCompat
 import com.facebook.react.modules.core.ReactChoreographer
 import com.facebook.react.uimanager.ThemedReactContext
 import com.facebook.react.uimanager.UIManagerHelper
+import com.facebook.react.views.view.ReactViewGroup
 import com.swmansion.rnscreens.Screen.ActivityState
 import com.swmansion.rnscreens.events.ScreenDismissedEvent
 
@@ -42,6 +44,7 @@ open class ScreenContainer(
             }
         }
     private var parentScreenWrapper: ScreenFragmentWrapper? = null
+    private var shouldBreakJSUpdateCount = 0
 
     override fun onLayout(
         changed: Boolean,
diff --git a/node_modules/react-native-screens/ios/RNSFullWindowOverlay.mm b/node_modules/react-native-screens/ios/RNSFullWindowOverlay.mm
index b7e3aaf..b4c638e 100644
--- a/node_modules/react-native-screens/ios/RNSFullWindowOverlay.mm
+++ b/node_modules/react-native-screens/ios/RNSFullWindowOverlay.mm
@@ -40,6 +40,29 @@ - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event
   return NO;
 }
 
+// Here is just copied from UIView+React, but we have modified `self.reactSubviews` to `self.subviews`
+// because `RNSFullWindowOverlayContainer` is just a simple UIView without the `reactSubviews` property.
+- (NSArray<UIView *> *)reactZIndexSortedSubviews
+{
+  // Check if sorting is required - in most cases it won't be.
+  BOOL sortingRequired = NO;
+  for (UIView *subview in self.subviews) {
+    if (subview.reactZIndex != 0) {
+      sortingRequired = YES;
+      break;
+    }
+  }
+  return sortingRequired ? [self.subviews sortedArrayUsingComparator:^NSComparisonResult(UIView *a, UIView *b) {
+    if (a.reactZIndex > b.reactZIndex) {
+      return NSOrderedDescending;
+    } else {
+      // Ensure sorting is stable by treating equal zIndex as ascending so
+      // that original order is preserved.
+      return NSOrderedAscending;
+    }
+  }] : self.subviews;
+}
+
 - (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
 {
   BOOL canReceiveTouchEvents = ([self isUserInteractionEnabled] && ![self isHidden]);
diff --git a/node_modules/react-native-screens/ios/RNSScreenStackAnimator.mm b/node_modules/react-native-screens/ios/RNSScreenStackAnimator.mm
index abb2cf6..3aaa91b 100644
--- a/node_modules/react-native-screens/ios/RNSScreenStackAnimator.mm
+++ b/node_modules/react-native-screens/ios/RNSScreenStackAnimator.mm
@@ -283,16 +283,27 @@ - (void)animateSlideFromBottomWithTransitionContext:(id<UIViewControllerContextT
   if (_operation == UINavigationControllerOperationPush) {
     toViewController.view.transform = topBottomTransform;
     [[transitionContext containerView] addSubview:toViewController.view];
-    [UIView animateWithDuration:[self transitionDuration:transitionContext]
-        animations:^{
-          fromViewController.view.transform = CGAffineTransformIdentity;
-          toViewController.view.transform = CGAffineTransformIdentity;
-        }
-        completion:^(BOOL finished) {
-          fromViewController.view.transform = CGAffineTransformIdentity;
-          toViewController.view.transform = CGAffineTransformIdentity;
-          [transitionContext completeTransition:![transitionContext transitionWasCancelled]];
-        }];
+    UIViewPropertyAnimator *modalAnimator = [[UIViewPropertyAnimator alloc] initWithDuration:[self transitionDuration:transitionContext] timingParameters:[[UISpringTimingParameters alloc] initWithMass:3 stiffness:1000 damping:500 initialVelocity:CGVectorMake(0, 0)]];
+    [modalAnimator addAnimations:^{
+      fromViewController.view.transform = CGAffineTransformIdentity;
+      toViewController.view.transform = CGAffineTransformIdentity;
+    }];
+    [modalAnimator addCompletion:^(UIViewAnimatingPosition finalPosition) {
+      fromViewController.view.transform = CGAffineTransformIdentity;
+      toViewController.view.transform = CGAffineTransformIdentity;
+      [transitionContext completeTransition:![transitionContext transitionWasCancelled]];
+    }];
+    [modalAnimator startAnimation];
+    // [UIView animateWithDuration:[self transitionDuration:transitionContext]
+    //     animations:^{
+    //       fromViewController.view.transform = CGAffineTransformIdentity;
+    //       toViewController.view.transform = CGAffineTransformIdentity;
+    //     }
+    //     completion:^(BOOL finished) {
+    //       fromViewController.view.transform = CGAffineTransformIdentity;
+    //       toViewController.view.transform = CGAffineTransformIdentity;
+    //       [transitionContext completeTransition:![transitionContext transitionWasCancelled]];
+    //     }];
   } else if (_operation == UINavigationControllerOperationPop) {
     toViewController.view.transform = CGAffineTransformIdentity;
     [[transitionContext containerView] insertSubview:toViewController.view belowSubview:fromViewController.view];
diff --git a/node_modules/react-native-screens/ios/RNSScreenStackHeaderConfig.mm b/node_modules/react-native-screens/ios/RNSScreenStackHeaderConfig.mm
index 63a09c1..3fdf5b2 100644
--- a/node_modules/react-native-screens/ios/RNSScreenStackHeaderConfig.mm
+++ b/node_modules/react-native-screens/ios/RNSScreenStackHeaderConfig.mm
@@ -158,7 +158,7 @@ - (void)updateViewControllerIfNeeded
   BOOL isInFullScreenModal = nav == nil && _screenView.stackPresentation == RNSScreenStackPresentationFullScreenModal;
   // if nav is nil, it means we can be in a fullScreen modal, so there is no nextVC, but we still want to update
   if (vc != nil && (nextVC == vc || isInFullScreenModal || isPresentingVC)) {
-    [RNSScreenStackHeaderConfig updateViewController:self.screenView.controller withConfig:self animated:YES];
+    [RNSScreenStackHeaderConfig updateViewController:self.screenView.controller withConfig:self animated:NO];
     // As the header might have change in `updateViewController` we need to ensure that header height
     // returned by the `onHeaderHeightChange` event is correct.
     [self.screenView.controller calculateAndNotifyHeaderHeightChangeIsModal:NO];
diff --git a/node_modules/react-native-screens/ios/UIWindow+RNScreens.mm b/node_modules/react-native-screens/ios/UIWindow+RNScreens.mm
index e295dcc..ea8d833 100644
--- a/node_modules/react-native-screens/ios/UIWindow+RNScreens.mm
+++ b/node_modules/react-native-screens/ios/UIWindow+RNScreens.mm
@@ -1,11 +1,20 @@
 #import "RNSFullWindowOverlay.h"
 #import "UIWindow+RNScreens.h"
+#import <React/RCTUtils.h>
 
 @implementation UIWindow (RNScreens)
 
 - (void)didAddSubview:(UIView *)subview
 {
   if (![subview isKindOfClass:[RNSFullWindowOverlayContainer class]]) {
+    // We only bring `RNSFullWindowOverlayContainer` above the `UIModal`; we don't handle other things like `SPAlert`
+    NSString *subviewClassString = NSStringFromClass([subview class]);
+    if (![subviewClassString hasPrefix:@"UI"] && ![subviewClassString hasPrefix:@"_UI"]) {
+      return;
+    }
+    if ([RCTPresentedViewController() isKindOfClass:[UIAlertController class]]) {
+      return;
+    }
     for (UIView *view in self.subviews) {
       if ([view isKindOfClass:[RNSFullWindowOverlayContainer class]]) {
         [self bringSubviewToFront:view];
diff --git a/node_modules/react-native-screens/lib/commonjs/components/Screen.web.js b/node_modules/react-native-screens/lib/commonjs/components/Screen.web.js
index af937e7..f657b27 100644
--- a/node_modules/react-native-screens/lib/commonjs/components/Screen.web.js
+++ b/node_modules/react-native-screens/lib/commonjs/components/Screen.web.js
@@ -16,6 +16,7 @@ const InnerScreen = exports.InnerScreen = _reactNative.View;
 // createAnimatedComponent` does not support stateless functional components; use a class component instead.
 // NOTE: React Server Components do not support class components.
 class NativeScreen extends _react.default.Component {
+  setNativeProps() {}
   render() {
     let {
       active,
diff --git a/node_modules/react-native-screens/lib/module/components/Screen.web.js b/node_modules/react-native-screens/lib/module/components/Screen.web.js
index e57ee93..cdd10fe 100644
--- a/node_modules/react-native-screens/lib/module/components/Screen.web.js
+++ b/node_modules/react-native-screens/lib/module/components/Screen.web.js
@@ -10,6 +10,7 @@ export const InnerScreen = View;
 // createAnimatedComponent` does not support stateless functional components; use a class component instead.
 // NOTE: React Server Components do not support class components.
 export class NativeScreen extends React.Component {
+  setNativeProps() {}
   render() {
     let {
       active,
diff --git a/node_modules/react-native-screens/src/components/Screen.web.tsx b/node_modules/react-native-screens/src/components/Screen.web.tsx
index 6b7b520..b7d42c2 100644
--- a/node_modules/react-native-screens/src/components/Screen.web.tsx
+++ b/node_modules/react-native-screens/src/components/Screen.web.tsx
@@ -12,6 +12,7 @@ export const InnerScreen = View;
 // createAnimatedComponent` does not support stateless functional components; use a class component instead.
 // NOTE: React Server Components do not support class components.
 export class NativeScreen extends React.Component<ScreenProps> {
+  setNativeProps() {}
   render(): JSX.Element {
     let {
       active,
