diff --git a/node_modules/react-native-logs/dist/transports/consoleTransport.js b/node_modules/react-native-logs/dist/transports/consoleTransport.js
index c27ec18..e0a5ae4 100644
--- a/node_modules/react-native-logs/dist/transports/consoleTransport.js
+++ b/node_modules/react-native-logs/dist/transports/consoleTransport.js
@@ -44,7 +44,7 @@ const consoleTransport = (props) => {
         msg = msg.replace(props.extension, `${extStart} ${props.extension} ${extEnd}`);
     }
     if ((_c = props.options) === null || _c === void 0 ? void 0 : _c.consoleFunc) {
-        props.options.consoleFunc(msg.trim());
+        props.options.consoleFunc(msg.trim(), props);
     }
     else {
         console.log(msg.trim());
diff --git a/node_modules/react-native-logs/dist/transports/fileAsyncTransport.js b/node_modules/react-native-logs/dist/transports/fileAsyncTransport.js
index a623bf8..dc65852 100644
--- a/node_modules/react-native-logs/dist/transports/fileAsyncTransport.js
+++ b/node_modules/react-native-logs/dist/transports/fileAsyncTransport.js
@@ -92,6 +92,8 @@ const fileAsyncTransport = (props) => {
     if ((_c = props === null || props === void 0 ? void 0 : props.options) === null || _c === void 0 ? void 0 : _c.filePath)
         filePath = props.options.filePath;
     let output = `${props === null || props === void 0 ? void 0 : props.msg}\n`;
+
+    filePath = filePath.replace(/\/$/,'');
     var path = filePath + "/" + fileName;
     WRITE(props.options.FS, path, output);
 };
