diff --git a/node_modules/react-native/.gradle/8.0.2/checksums/checksums.lock b/node_modules/react-native/.gradle/8.0.2/checksums/checksums.lock
new file mode 100644
index 0000000..f14a8c8
Binary files /dev/null and b/node_modules/react-native/.gradle/8.0.2/checksums/checksums.lock differ
diff --git a/node_modules/react-native/.gradle/8.0.2/checksums/sha1-checksums.bin b/node_modules/react-native/.gradle/8.0.2/checksums/sha1-checksums.bin
new file mode 100644
index 0000000..3e1fca4
Binary files /dev/null and b/node_modules/react-native/.gradle/8.0.2/checksums/sha1-checksums.bin differ
diff --git a/node_modules/react-native/.gradle/8.0.2/dependencies-accessors/dependencies-accessors.lock b/node_modules/react-native/.gradle/8.0.2/dependencies-accessors/dependencies-accessors.lock
new file mode 100644
index 0000000..1a0075b
Binary files /dev/null and b/node_modules/react-native/.gradle/8.0.2/dependencies-accessors/dependencies-accessors.lock differ
diff --git a/node_modules/react-native/.gradle/8.0.2/dependencies-accessors/gc.properties b/node_modules/react-native/.gradle/8.0.2/dependencies-accessors/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native/.gradle/8.0.2/fileChanges/last-build.bin b/node_modules/react-native/.gradle/8.0.2/fileChanges/last-build.bin
new file mode 100644
index 0000000..f76dd23
Binary files /dev/null and b/node_modules/react-native/.gradle/8.0.2/fileChanges/last-build.bin differ
diff --git a/node_modules/react-native/.gradle/8.0.2/fileHashes/fileHashes.lock b/node_modules/react-native/.gradle/8.0.2/fileHashes/fileHashes.lock
new file mode 100644
index 0000000..80940a1
Binary files /dev/null and b/node_modules/react-native/.gradle/8.0.2/fileHashes/fileHashes.lock differ
diff --git a/node_modules/react-native/.gradle/8.0.2/gc.properties b/node_modules/react-native/.gradle/8.0.2/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native/.gradle/buildOutputCleanup/buildOutputCleanup.lock b/node_modules/react-native/.gradle/buildOutputCleanup/buildOutputCleanup.lock
new file mode 100644
index 0000000..d76d4e1
Binary files /dev/null and b/node_modules/react-native/.gradle/buildOutputCleanup/buildOutputCleanup.lock differ
diff --git a/node_modules/react-native/.gradle/buildOutputCleanup/cache.properties b/node_modules/react-native/.gradle/buildOutputCleanup/cache.properties
new file mode 100644
index 0000000..87e5e64
--- /dev/null
+++ b/node_modules/react-native/.gradle/buildOutputCleanup/cache.properties
@@ -0,0 +1,2 @@
+#Fri Mar 29 15:25:21 CST 2024
+gradle.version=8.0.2
diff --git a/node_modules/react-native/.gradle/buildOutputCleanup/outputFiles.bin b/node_modules/react-native/.gradle/buildOutputCleanup/outputFiles.bin
new file mode 100644
index 0000000..0d7059f
Binary files /dev/null and b/node_modules/react-native/.gradle/buildOutputCleanup/outputFiles.bin differ
diff --git a/node_modules/react-native/.gradle/vcs-1/gc.properties b/node_modules/react-native/.gradle/vcs-1/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native/.gradle/workspace-id.txt b/node_modules/react-native/.gradle/workspace-id.txt
new file mode 100644
index 0000000..510e1bd
Binary files /dev/null and b/node_modules/react-native/.gradle/workspace-id.txt differ
diff --git a/node_modules/react-native/.gradle/workspace-id.txt.lock b/node_modules/react-native/.gradle/workspace-id.txt.lock
new file mode 100644
index 0000000..97a22cb
Binary files /dev/null and b/node_modules/react-native/.gradle/workspace-id.txt.lock differ
diff --git a/node_modules/react-native/Libraries/Components/View/ViewAccessibility.d.ts b/node_modules/react-native/Libraries/Components/View/ViewAccessibility.d.ts
index b06d1cb..6c2e083 100644
--- a/node_modules/react-native/Libraries/Components/View/ViewAccessibility.d.ts
+++ b/node_modules/react-native/Libraries/Components/View/ViewAccessibility.d.ts
@@ -97,7 +97,7 @@ export interface AccessibilityProps
    * A value indicating whether the accessibility elements contained within
    * this accessibility element are hidden.
    */
-  'aria-hidden'?: boolean | undefined;
+  'aria-hidden'?: boolean | undefined | string;
 
   'aria-live'?: ('polite' | 'assertive' | 'off') | undefined;
   'aria-modal'?: boolean | undefined;
diff --git a/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js b/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js
index 6c7116e..eb32fdc 100644
--- a/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js
+++ b/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js
@@ -4333,7 +4333,10 @@ function injectInternals(internals) {
     }
 
     rendererID = hook.inject(internals); // We have successfully injected, so now it is safe to set up hooks.
-
+    // For support react-render-tracker on iOS
+    globalThis.__REACT_DEVTOOLS_GLOBAL_INTERNALS__ = internals;
+    globalThis.__REACT_DEVTOOLS_GLOBAL_INTERNALS_RENDERER_ID__  = rendererID;
+     injectedHook = hook;
     injectedHook = hook;
   } catch (err) {
     // Catch all errors because it is unsafe to throw during initialization.
diff --git a/node_modules/react-native/ReactCommon/jsc/JSCRuntime.cpp b/node_modules/react-native/ReactCommon/jsc/JSCRuntime.cpp
index 38ebeda..4d6142f 100644
--- a/node_modules/react-native/ReactCommon/jsc/JSCRuntime.cpp
+++ b/node_modules/react-native/ReactCommon/jsc/JSCRuntime.cpp
@@ -417,13 +417,16 @@ JSCRuntime::~JSCRuntime() {
   // has started.
   ctxInvalid_ = true;
   JSGlobalContextRelease(ctx_);
-#ifndef NDEBUG
-  assert(
-      objectCounter_ == 0 && "JSCRuntime destroyed with a dangling API object");
-  assert(
-      stringCounter_ == 0 && "JSCRuntime destroyed with a dangling API string");
-#endif
-}
+  
+// fix issue that app crashes on reload in development mode with JSC engine. 
+// details: https://github.com/software-mansion/react-native-reanimated/issues/4783
+// #ifndef NDEBUG
+//   assert(
+//       objectCounter_ == 0 && "JSCRuntime destroyed with a dangling API object");
+//   assert(
+//       stringCounter_ == 0 && "JSCRuntime destroyed with a dangling API string");
+// #endif
+// }
 
 std::shared_ptr<const jsi::PreparedJavaScript> JSCRuntime::prepareJavaScript(
     const std::shared_ptr<const jsi::Buffer> &buffer,
