diff --git a/node_modules/react-native/sdks/.hermesversion b/node_modules/react-native/sdks/.hermesversion
index b235e48..6b1ffcf 100644
--- a/node_modules/react-native/sdks/.hermesversion
+++ b/node_modules/react-native/sdks/.hermesversion
@@ -1 +1 @@
-hermes-2023-03-07-RNv0.71.4-31fdcf738940875c9bacf251e149006cf515d763
\ No newline at end of file
+ae2ac688c3faf868d2360a10a7791226ed5702fb
\ No newline at end of file
