diff --git a/node_modules/react-native-bundle-visualizer/.DS_Store b/node_modules/react-native-bundle-visualizer/.DS_Store
new file mode 100644
index 0000000..379b59a
Binary files /dev/null and b/node_modules/react-native-bundle-visualizer/.DS_Store differ
diff --git a/node_modules/react-native-bundle-visualizer/src/react-native-bundle-visualizer.js b/node_modules/react-native-bundle-visualizer/src/react-native-bundle-visualizer.js
index 6487b5c..fc1d986 100755
--- a/node_modules/react-native-bundle-visualizer/src/react-native-bundle-visualizer.js
+++ b/node_modules/react-native-bundle-visualizer/src/react-native-bundle-visualizer.js
@@ -67,7 +67,7 @@ const bundleOutput =
   argv['bundle-output'] || path.join(tmpDir, platform + '.bundle');
 const bundleOutputSourceMap = bundleOutput + '.map';
 const format = argv.format || 'html';
-const bundleOutputExplorerFile = path.join(outDir, 'explorer.' + format);
+const bundleOutputExplorerFile = 'stats.' + format;
 const onlyMapped = !!argv['only-mapped'] || false;
 
 // Make sure the temp dir exists
@@ -187,5 +187,5 @@ bundlePromise
     }
 
     // Open output file
-    return open(bundleOutputExplorerFile);
+    if(format==='html')open(bundleOutputExplorerFile);
   });
