diff --git a/node_modules/react-native/Libraries/Components/TextInput/AndroidTextInputNativeComponent.js b/node_modules/react-native/Libraries/Components/TextInput/AndroidTextInputNativeComponent.js
index 55b770d..8f463db 100644
--- a/node_modules/react-native/Libraries/Components/TextInput/AndroidTextInputNativeComponent.js
+++ b/node_modules/react-native/Libraries/Components/TextInput/AndroidTextInputNativeComponent.js
@@ -750,7 +750,7 @@ export const __INTERNAL_VIEW_CONFIG: PartialViewConfig = {
 };
 
 let AndroidTextInputNativeComponent = NativeComponentRegistry.get<NativeProps>(
-  'AndroidTextInput',
+  'AndroidPasteTextInput',
   () => __INTERNAL_VIEW_CONFIG,
 );
 
