diff --git a/node_modules/react-native-vision-camera/ios/React Utils/MakeJSIRuntime.h b/node_modules/react-native-vision-camera/ios/React Utils/MakeJSIRuntime.h
index a428c01..2319316 100644
--- a/node_modules/react-native-vision-camera/ios/React Utils/MakeJSIRuntime.h	
+++ b/node_modules/react-native-vision-camera/ios/React Utils/MakeJSIRuntime.h	
@@ -20,8 +20,11 @@
 #elif __has_include(<v8runtime/V8RuntimeFactory.h>)
   // V8 (https://github.com/Kudo/react-native-v8)
   #include <v8runtime/V8RuntimeFactory.h>
+#elif __has_include(<React-jsc/JSCRuntime.h>)
+  // JSC with Hermes disabled
+  #include <React-jsc/JSCRuntime.h>
 #else
-  // JSC
+  // JSC with Hermes enabled
   #include <jsi/JSCRuntime.h>
 #endif
 
