diff --git a/node_modules/realm/binding/android/CMakeLists.txt b/node_modules/realm/binding/android/CMakeLists.txt
index 87af51f..136a566 100644
--- a/node_modules/realm/binding/android/CMakeLists.txt
+++ b/node_modules/realm/binding/android/CMakeLists.txt
@@ -29,7 +29,12 @@ set_target_properties(realm-js-android-binding PROPERTIES
 set(CMAKE_INTERPROCEDURAL_OPTIMIZATION OFF)
 set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG -Oz")
 
-target_link_options(realm-js-android-binding PUBLIC -fvisibility=hidden)
+target_link_options(realm-js-android-binding PUBLIC
+    -fvisibility=hidden
+    -Wl,-z,max-page-size=16384
+    -Wl,--hash-style=both
+    -Wl,--build-id
+)
 
 target_include_directories(realm-js-android-binding PRIVATE
     "${CMAKE_CURRENT_SOURCE_DIR}/.."
diff --git a/node_modules/realm/binding/android/build.gradle b/node_modules/realm/binding/android/build.gradle
index 203737e..7972e88 100644
--- a/node_modules/realm/binding/android/build.gradle
+++ b/node_modules/realm/binding/android/build.gradle
@@ -57,6 +57,7 @@ android {
             cmake {
                 arguments "-DANDROID_STL=c++_shared",
                           "-DCMAKE_CXX_VISIBILITY_PRESET=hidden",
+                          "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
                           "-DREACT_NATIVE_ROOT_DIR=${REACT_NATIVE_ROOT_DIR}"
                 targets 'realm-js-android-binding'
                 cppFlags '-O2 -std=c++20 -frtti -fexceptions -Wall -fstack-protector-all'
@@ -81,7 +82,8 @@ android {
         release {
             externalNativeBuild {
                 cmake {
-                    arguments "-DCMAKE_BUILD_TYPE=Release"
+                    arguments "-DCMAKE_BUILD_TYPE=Release",
+                        "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
                 }
             }
         }
