diff --git a/node_modules/react-native/Libraries/StyleSheet/StyleSheet.js b/node_modules/react-native/Libraries/StyleSheet/StyleSheet.js
index 785fbfe..7446069 100644
--- a/node_modules/react-native/Libraries/StyleSheet/StyleSheet.js
+++ b/node_modules/react-native/Libraries/StyleSheet/StyleSheet.js
@@ -347,7 +347,7 @@ module.exports = {
     }
 
     if (__DEV__ && typeof value.process === 'function') {
-      console.warn(`Overwriting ${property} style attribute preprocessor`);
+      // console.warn(`Overwriting ${property} style attribute preprocessor`);
     }
 
     ReactNativeStyleAttributes[property] = value;
